### ISSUE TITLE
Change Request: magic comment `eslint-expect-error-next-line`

### ISSUE THREAD

**Issue by loynoir** [eslint/eslint#15709]
### ESLint version

v8.11.0

### What problem do you want to solve?

Given 
1. eslint config, for example, have rule `no-alert`
2. some unit test code for eslint config, for example
```js
alert('foo');
```
How to write test againt eslint config easily?

I did some search, seems no luck.

So, request a new magic comment.

### What do you think is the correct solution?

Similar to typescript `// @ts-expect-error`

`// eslint-expect-error-next-line [X[, Y [, ...]]]`

1. OK, when trigger `no-alert`
```js
// eslint-expect-error-next-line no-alert
alert('foo');
```

2. Error, when not trigger `no-alert`
```js
// eslint-expect-error-next-line no-alert
42;
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @loynoir, thanks for the issue!

> Given
> 
> 1. eslint config, for example, have rule `no-alert`
> 2. some unit test code for eslint config, for example
> 
> ```js
> alert('foo');
> ```
> 
> How to write test againt eslint config easily?

I think the [`--report-unused-disable-directives`](https://eslint.org/docs/user-guide/command-line-interface#--report-unused-disable-directives) CLI flag can be useful for this use case.

For example, if you add an `eslint-disable` directive:

```js
//------------- test.js ----------------------

// eslint-disable-next-line no-alert
alert('foo');
```

and run eslint on the test file:

```
npx eslint test.js --report-unused-disable-directives
```

It will report an error and exit with 1 if your configuration doesn't have the `no-alert` rule enabled.

Would that work for you?

**Comment by loynoir** [comment]
@mdjermanovic Thanks, it works in many cases.

But sometimes, same logic refers to multiple names.

### Question
Is there a `eslint-expect-error-next-line A | B`,
which expect next line throw error A or throw error B?

### Example

`// eslint-expect-error-next-line no-unused-vars | @typescript-eslint/no-unused-vars`

```js
const unused = 2
```

`// eslint-expect-error-next-line semi | prettier/prettier`
```js
console.log(1);
```

**Comment by mdjermanovic** [comment]
> Is there a `eslint-expect-error-next-line A | B`,
> which expect next line throw error A or throw error B?

No, that functionality doesn't exist in ESLint. For checks where `eslint-disable` with `--report-unused-disable-directives` doesn't do the work, you could write tests that run ESLint through the [API](https://eslint.org/docs/developer-guide/nodejs-api) and assert the results.

```js
const assert = require("assert");
const { ESLint } = require("eslint");

describe("My ESLint config", () => {
    const eslint = new ESLint({
        overrideConfigFile: "my-eslint-config.js",
        useEslintrc: false,
    });

    it("should report unused vars", async () => {
        const [{ messages }] = await eslint.lintText("const unused = 2");
        assert(
            messages.some(({ ruleId }) =>
                [
                    "no-unused-vars",
                    "@typescript-eslint/no-unused-vars",
                ].includes(ruleId)
            )
        );
    });
});

```

 I’m curious what other team members think about supporting directives that would allow for testing that is more complex than what can be achieved with existing directives.

**Comment by nzakas** [comment]
I’m not in favor of this proposal. I think you can get most of what you’re looking for with a combination of disable comments and reporting unused disable directives. I don’t think adding a new directive type provides enough value for the additional complexity.

**Comment by mdjermanovic** [comment]
I also don't think this provides enough value for additional complexity. `eslint-disable` comments combined with `--report-unused-disable-directives` cover most use cases, and for those not covered there is always the option to test configurations with the ESLint API as described in https://github.com/eslint/eslint/issues/15709#issuecomment-1074113452.

Since there’s not much interest from the team in moving forward here, I'm closing this issue.

**Comment by loynoir** [comment]
Oh, I found something similar.

No need to write long tests. :)

https://mysticatea.github.io/eslint-plugin-eslint-comments/
